feature-libs/pickup-in-store/core/store/effects/pickup-location.effect.spec.ts

```diff
@@ -0,0 +1,164 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { TestBed } from '@angular/core/testing';
+import { provideMockActions } from '@ngrx/effects/testing';
+import { StoreModule } from '@ngrx/store';
+import { normalizeHttpError, PointOfService } from '@spartacus/core';
+import {
+  SetPickupOptionDeliveryPayload,
+  SetPickupOptionInStorePayload,
+} from '@spartacus/pickup-in-store/root';
+import { cold, hot } from 'jasmine-marbles';
+import { Observable, of } from 'rxjs';
+import { PickupLocationConnector } from '../../connectors';
+import {
+  GetStoreDetailsById,
+  SetStoreDetailsFailure,
+  SetStoreDetailsSuccess,
+  SetPickupOptionDelivery,
+  SetPickupOptionDeliverySuccess,
+  SetPickupOptionInStore,
+  SetPickupOptionInStoreSuccess,
+} from '../actions/pickup-location.action';
+import { PickupLocationEffect } from './pickup-location.effect';
+
+class MockPickupLocationConnector {
+  getStoreDetails(_storeName: string): Observable<PointOfService> {
+    return of({});
+  }
+  setPickupOptionDelivery(
+    _cartId: string,
+    _entryNumber: number,
+    _userId: string,
+    _requestPayload: any
+  ): Observable<any> {
+    return of({});
+  }
+  setPickupOptionInStore(
+    _cartId: string,
+    _entryNumber: number,
+    _userId: string,
+    _requestPayload: any
+  ): Observable<any> {
+    return of({});
+  }
+}
```

Comment:
I'd prefer if re-usable mocks like this lived with the tests for the thing that they are mocking. It is easier to find and re-use them that way.